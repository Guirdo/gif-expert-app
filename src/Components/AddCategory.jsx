import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Form } from 'react-bootstrap';

const AddCategory = ({setCategories}) => {

    const [inputValue,setInputValue] = useState('');
    
    const handleInputChange = (e) =>{
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) =>{
        e.preventDefault();

        if(inputValue.trim().length > 2){
            setCategories(c => [inputValue,...c]);
            setInputValue('');
        }

    }

    return (
        <Form onSubmit={handleSubmit}>
            <Form.Control 
                type="text"
                value={inputValue}
                placeholder="Search anything you want"
                onChange={ handleInputChange }
            />
        </Form>
    )
}

AddCategory.propTypes ={
    setCategories: PropTypes.func.isRequired,
}

export default AddCategory;
