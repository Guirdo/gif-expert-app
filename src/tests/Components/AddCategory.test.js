import { shallow } from "enzyme";
import AddCategory from "../../Components/AddCategory";

describe('Testing AddCategory',()=>{

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach(()=>{
        jest.clearAllMocks();

        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });

    test('should show it right',()=>{
        expect(wrapper).toMatchSnapshot();
    });

    /* test('should change the input field',()=>{
        const input = wrapper.find('FormControl');
        const value = 'Hola mundo';

        input.simulate('change',{ target: {value} });
    }); */

    test('shouldnt post on submit',()=>{
        wrapper.find('Form').simulate('submit',{ preventDefault(){} });

        expect(setCategories).not.toHaveBeenCalled();
    });

    test('should call setCategories and clean the input',()=>{
        const input = wrapper.find('FormControl');
        const value = 'Hola mundo';

        input.simulate('change',{ target: {value} });

        wrapper.find('Form').simulate('submit',{ preventDefault(){} });

        expect(setCategories).toHaveBeenCalled();
        expect(input.prop('value')).toBe('');
    });

});