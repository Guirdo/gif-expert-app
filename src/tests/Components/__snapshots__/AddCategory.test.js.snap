// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddCategory should show it right 1`] = `
<Form
  onSubmit={[Function]}
>
  <FormControl
    onChange={[Function]}
    placeholder="Search anything you want"
    type="text"
    value=""
  />
</Form>
`;
